core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -678,6 +707,24 @@ private ConnectorExpression constantFor(Expression node)
             return Optional.empty();
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitIfExpression(IfExpression node, Void context)
+        {
+            Optional<ConnectorExpression> firstValue = process(node.getCondition());
+            Optional<ConnectorExpression> trueValue = process(node.getTrueValue());
+            if (firstValue.isPresent() && trueValue.isPresent()) {
+                if (node.getFalseValue().isEmpty()) {
+                    return Optional.of(new Call(typeOf(node), IF_FUNCTION_NAME, ImmutableList.of(firstValue.get(), trueValue.get())));
```

Comment:
I talked with @findepi about this, we're not sure if we should translate to `$if` or to `$case` as a private case. WDYT?

Reply:
What does "a private case" mean here?